jsk_pcl_ros_utils/src/polygon_array_likelihood_filter_nodelet.cpp

```diff
@@ -0,0 +1,166 @@
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/o2r other materials provided
+ *     with the distribution.
+ *   * Neither the name of the JSK Lab nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+#include <algorithm>
+#include <jsk_pcl_ros_utils/polygon_array_likelihood_filter.h>
+
+namespace jsk_pcl_ros_utils
+{
+  void PolygonArrayLikelihoodFilter::onInit()
+  {
+    DiagnosticNodelet::onInit();
+
+    srv_ = boost::make_shared<dynamic_reconfigure::Server<Config> >(*pnh_);
+    dynamic_reconfigure::Server<Config>::CallbackType f =
+      boost::bind(&PolygonArrayLikelihoodFilter::configCallback, this, _1, _2);
+    srv_->setCallback(f);
+
+    pub_polygons_ = advertise<jsk_recognition_msgs::PolygonArray>(
+      *pnh_, "output_polygons", 1);
+
+    pnh_->param<bool>(std::string("use_coefficients"), use_coefficients_, true);
+    if (use_coefficients_) {
+      pub_coefficients_ = advertise<jsk_recognition_msgs::ModelCoefficientsArray>(
+        *pnh_, "output_coefficients", 1);
+    }
+
+    onInitPostProcess();
+  }
+
+  void PolygonArrayLikelihoodFilter::subscribe()
+  {
+    if (use_coefficients_) {
+      sync_ = boost::make_shared<message_filters::Synchronizer<SyncPolicy> >(queue_size_);
+      sub_polygons_.subscribe(*pnh_, "input_polygons", 1);
+      sub_coefficients_.subscribe(*pnh_, "input_coefficients", 1);
+      sync_->connectInput(sub_polygons_, sub_coefficients_);
+      sync_->registerCallback(boost::bind(
+                                &PolygonArrayLikelihoodFilter::filter,
+                                this, _1, _2));
+    } else {
+      sub_polygons_alone_ = pnh_->subscribe(
+        "input_polygons", 1, &PolygonArrayLikelihoodFilter::filter, this);
```

Comment:
@wkentaro Why this is not allowed? `sub_polygons_alone` is just a `ros::Suscriber` not a `message_filters::Subscriber`.